pyk/src/pyk/kcfg/kcfg.py

```diff
@@ -1040,6 +1040,24 @@ def create_split(self, source_id: NodeIdLike, splits: Iterable[tuple[NodeIdLike,
         self.add_successor(split)
         return split
 
+    def create_split_by_nodes(self, source_id: NodeIdLike, target_ids: Iterable[NodeIdLike]) -> KCFG.Split | None:
+        """Create a split without crafting a CSubst."""
+        source = self.node(source_id)
+        targets = [self.node(nid) for nid in target_ids]
+        csubsts: list[CSubst] = []
+
+        for target in targets:
+            subst = source.cterm.config.match(target.cterm.config)
+            if subst is None:
+                return None
+            source_constraints = [subst(c) for c in source.cterm.constraints]
+            constraints = [c for c in target.cterm.constraints if c not in source_constraints] + [
+                c for c in source_constraints if c not in target.cterm.constraints
+            ]
+            csubsts.append(CSubst(subst, constraints))
```

Comment:
What if instead we use `CTerm.match_with_constraint`: https://github.com/runtimeverification/k/blob/42e03545c319847e7df84e6fbc196b2f51c148a0/pyk/src/pyk/cterm/cterm.py#L162

```suggestion
        csubsts = [not_none(source.cterm.match_with_constraint(target.cterm)) for target in targets]
```

You can import `not_none` from `..utils`.

Reply:
`CTerm.match_with_constraint` doesn't give the same result as expected. It always returns a true imply as the constraints for CSubst. That's why we have #4499 and #4602. Since `match_with_constraints` is already in use, modifying it might cause downstream issues. Additionally, I'd like to provide merge node functionality as soon as possible. So, I give up refactoring `match_with_constraints` and instead provide this implementation here.

After we have the result of the discussion at #4602, we can refactor `match_with_constraints` and `create_split_by_nodes`.

Reply:
Please leave a comment here saying that when #4496 is closed (link to it) that we can change this to use `match_with_constarint`

Reply:
Done!